### ISSUE TITLE
Package install fail post fresh-install

### ISSUE THREAD

**Issue by George9000** [syl20bnr/spacemacs#5513]
Emacs 25.1.50.1
spacemacs-0.105.14
OSX 10.11.3
Launching Emacs with:
  open -a /Applications/Emacs.app --args --debug-init --insecure

After a new install by removing the ~/.emacs.d directory, attempting to add a single layer, org, gives this error. Similar error after checking out version 0.105.13. Any new layer install fails similarly.

```
Debugger entered: ((cl-assertion-failed (or (= (buffer-size tar-data-buffer) (buffer-size)) (eq tar-data-swapped (> (buffer-size tar-data-buffer) (buffer-size)))) nil))
  cl--assertion-failed((or (= (buffer-size tar-data-buffer) (buffer-size)) (eq tar-data-swapped (> (buffer-size tar-data-buffer) (buffer-size)))))
  tar-data-swapped-p()
  tar-untar-buffer()
  package-untar-buffer("gnuplot-20141231.1337")
  package-unpack([cl-struct-package-desc gnuplot (20141231 1337) "drive gnuplot from within emacs" nil tar "melpa" nil ((:keywords "gnuplot" "plotting")) nil])
  package-install-from-archive([cl-struct-package-desc gnuplot (20141231 1337) "drive gnuplot from within emacs" nil tar "melpa" nil ((:keywords "gnuplot" "plotting")) nil])
  mapc(package-install-from-archive ([cl-struct-package-desc gnuplot (20141231 1337) "drive gnuplot from within emacs" nil tar "melpa" nil ((:keywords "gnuplot" "plotting")) nil]))
  package-download-transaction(([cl-struct-package-desc gnuplot (20141231 1337) "drive gnuplot from within emacs" nil tar "melpa" nil ((:keywords "gnuplot" "plotting")) nil]))
  package-install(gnuplot)
  (if (not (assq pkg-name package-archive-contents)) (spacemacs-buffer/append (format (concat "\nPackage %s is unavailable. " "Is the package name misspelled?\n") pkg-name)) (let ((--dolist-tail-- (configuration-layer//get-package-deps-from-archive pkg-name)) dep) (while --dolist-tail-- (setq dep (car --dolist-tail--)) (if (package-installed-p (car dep)) (configuration-layer//activate-package (car dep)) (package-install (car dep))) (setq --dolist-tail-- (cdr --dolist-tail--)))) (package-install pkg-name))
  configuration-layer//install-from-elpa(gnuplot)
  (cond ((or (null pkg) (eq (quote elpa) location)) (configuration-layer//install-from-elpa pkg-name)) ((and (listp location) (eq (quote recipe) (car location))) (configuration-layer//install-from-recipe pkg)) (t (spacemacs-buffer/warning "Cannot install package %S." pkg-name)))
  (condition-case err (cond ((or (null pkg) (eq (quote elpa) location)) (configuration-layer//install-from-elpa pkg-name)) ((and (listp location) (eq (quote recipe) (car location))) (configuration-layer//install-from-recipe pkg)) (t (spacemacs-buffer/warning "Cannot install package %S." pkg-name))) ((quote error) (configuration-layer//increment-error-count) (spacemacs-buffer/append (format (concat "\nAn error occurred while installing %s " "(error: %s)\n") pkg-name err))))
  (if (package-installed-p pkg-name) nil (condition-case err (cond ((or (null pkg) (eq (quote elpa) location)) (configuration-layer//install-from-elpa pkg-name)) ((and (listp location) (eq (quote recipe) (car location))) (configuration-layer//install-from-recipe pkg)) (t (spacemacs-buffer/warning "Cannot install package %S." pkg-name))) ((quote error) (configuration-layer//increment-error-count) (spacemacs-buffer/append (format (concat "\nAn error occurred while installing %s " "(error: %s)\n") pkg-name err)))))
  (let* ((pkg (object-assoc pkg-name :name configuration-layer--packages)) (layer (if pkg (progn (eieio-oref pkg (quote :owner))))) (location (if pkg (progn (eieio-oref pkg (quote :location)))))) (spacemacs-buffer/replace-last-line (format "--> installing %s: %s%s... [%s/%s]" (if layer "package" "dependency") pkg-name (if layer (format "@%S" layer) "") installed-count noinst-count) t) (if (package-installed-p pkg-name) nil (condition-case err (cond ((or (null pkg) (eq (quote elpa) location)) (configuration-layer//install-from-elpa pkg-name)) ((and (listp location) (eq (quote recipe) (car location))) (configuration-layer//install-from-recipe pkg)) (t (spacemacs-buffer/warning "Cannot install package %S." pkg-name))) ((quote error) (configuration-layer//increment-error-count) (spacemacs-buffer/append (format (concat "\nAn error occurred while installing %s " "(error: %s)\n") pkg-name err))))))
  (while --dolist-tail-- (setq pkg-name (car --dolist-tail--)) (setq installed-count (1+ installed-count)) (let* ((pkg (object-assoc pkg-name :name configuration-layer--packages)) (layer (if pkg (progn (eieio-oref pkg (quote :owner))))) (location (if pkg (progn (eieio-oref pkg (quote :location)))))) (spacemacs-buffer/replace-last-line (format "--> installing %s: %s%s... [%s/%s]" (if layer "package" "dependency") pkg-name (if layer (format "@%S" layer) "") installed-count noinst-count) t) (if (package-installed-p pkg-name) nil (condition-case err (cond ((or (null pkg) (eq ... location)) (configuration-layer//install-from-elpa pkg-name)) ((and (listp location) (eq ... ...)) (configuration-layer//install-from-recipe pkg)) (t (spacemacs-buffer/warning "Cannot install package %S." pkg-name))) ((quote error) (configuration-layer//increment-error-count) (spacemacs-buffer/append (format (concat "\nAn error occurred while installing %s " "(error: %s)\n") pkg-name err)))))) (spacemacs//redisplay) (setq --dolist-tail-- (cdr --dolist-tail--)))
  (let ((--dolist-tail-- noinst-pkg-names) pkg-name) (while --dolist-tail-- (setq pkg-name (car --dolist-tail--)) (setq installed-count (1+ installed-count)) (let* ((pkg (object-assoc pkg-name :name configuration-layer--packages)) (layer (if pkg (progn (eieio-oref pkg ...)))) (location (if pkg (progn (eieio-oref pkg ...))))) (spacemacs-buffer/replace-last-line (format "--> installing %s: %s%s... [%s/%s]" (if layer "package" "dependency") pkg-name (if layer (format "@%S" layer) "") installed-count noinst-count) t) (if (package-installed-p pkg-name) nil (condition-case err (cond ((or ... ...) (configuration-layer//install-from-elpa pkg-name)) ((and ... ...) (configuration-layer//install-from-recipe pkg)) (t (spacemacs-buffer/warning "Cannot install package %S." pkg-name))) ((quote error) (configuration-layer//increment-error-count) (spacemacs-buffer/append (format ... pkg-name err)))))) (spacemacs//redisplay) (setq --dolist-tail-- (cdr --dolist-tail--))))
  (progn (spacemacs-buffer/append (format "Found %s new package(s) to install...\n" noinst-count)) (configuration-layer/retrieve-package-archives) (setq installed-count 0) (let ((--dolist-tail-- noinst-pkg-names) pkg-name) (while --dolist-tail-- (setq pkg-name (car --dolist-tail--)) (setq installed-count (1+ installed-count)) (let* ((pkg (object-assoc pkg-name :name configuration-layer--packages)) (layer (if pkg (progn ...))) (location (if pkg (progn ...)))) (spacemacs-buffer/replace-last-line (format "--> installing %s: %s%s... [%s/%s]" (if layer "package" "dependency") pkg-name (if layer (format "@%S" layer) "") installed-count noinst-count) t) (if (package-installed-p pkg-name) nil (condition-case err (cond (... ...) (... ...) (t ...)) ((quote error) (configuration-layer//increment-error-count) (spacemacs-buffer/append ...))))) (spacemacs//redisplay) (setq --dolist-tail-- (cdr --dolist-tail--)))) (spacemacs-buffer/append "\n"))
  (if noinst-pkg-names (progn (spacemacs-buffer/append (format "Found %s new package(s) to install...\n" noinst-count)) (configuration-layer/retrieve-package-archives) (setq installed-count 0) (let ((--dolist-tail-- noinst-pkg-names) pkg-name) (while --dolist-tail-- (setq pkg-name (car --dolist-tail--)) (setq installed-count (1+ installed-count)) (let* ((pkg (object-assoc pkg-name :name configuration-layer--packages)) (layer (if pkg ...)) (location (if pkg ...))) (spacemacs-buffer/replace-last-line (format "--> installing %s: %s%s... [%s/%s]" (if layer "package" "dependency") pkg-name (if layer ... "") installed-count noinst-count) t) (if (package-installed-p pkg-name) nil (condition-case err (cond ... ... ...) (... ... ...)))) (spacemacs//redisplay) (setq --dolist-tail-- (cdr --dolist-tail--)))) (spacemacs-buffer/append "\n")))
  (let* ((noinst-pkg-names (configuration-layer//get-uninstalled-packages (mapcar (quote car) (object-assoc-list :name configuration-layer--used-distant-packages)))) (noinst-count (length noinst-pkg-names)) installed-count) (if noinst-pkg-names (progn (spacemacs-buffer/append (format "Found %s new package(s) to install...\n" noinst-count)) (configuration-layer/retrieve-package-archives) (setq installed-count 0) (let ((--dolist-tail-- noinst-pkg-names) pkg-name) (while --dolist-tail-- (setq pkg-name (car --dolist-tail--)) (setq installed-count (1+ installed-count)) (let* ((pkg ...) (layer ...) (location ...)) (spacemacs-buffer/replace-last-line (format "--> installing %s: %s%s... [%s/%s]" ... pkg-name ... installed-count noinst-count) t) (if (package-installed-p pkg-name) nil (condition-case err ... ...))) (spacemacs//redisplay) (setq --dolist-tail-- (cdr --dolist-tail--)))) (spacemacs-buffer/append "\n"))))
  configuration-layer//install-packages(([eieio-class-tag--cfgl-package ace-jump-helm-line spacemacs nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package ace-link spacemacs nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package ace-window spacemacs nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package adaptive-wrap spacemacs nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package aggressive-indent spacemacs nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package auto-compile emacs-lisp nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package auto-highlight-symbol spacemacs nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package avy spacemacs nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package bind-key spacemacs-base nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package bind-map spacemacs-base nil nil elpa pre nil nil] [eieio-class-tag--cfgl-package bookmark spacemacs-base nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package bracketed-paste spacemacs nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package buffer-move spacemacs nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package centered-cursor spacemacs nil nil local nil nil nil] [eieio-class-tag--cfgl-package clean-aindent-mode spacemacs nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package company nil nil (emacs-lisp) elpa nil nil nil] [eieio-class-tag--cfgl-package company-emoji nil nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package define-word spacemacs nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package desktop spacemacs nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package diminish spacemacs-base nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package doc-view spacemacs nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package ediff spacemacs-base nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package eldoc spacemacs-base nil (emacs-lisp) elpa nil nil nil] [eieio-class-tag--cfgl-package electric-indent-mode spacemacs-base nil nil built-in nil nil nil] [eieio-class-tag--cfgl-package elisp-slime-nav emacs-lisp nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package emacs-lisp emacs-lisp nil nil built-in nil nil nil] [eieio-class-tag--cfgl-package emoji-cheat-sheet-plus nil nil (org) elpa nil nil nil] [eieio-class-tag--cfgl-package eval-sexp-fu spacemacs nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package evil spacemacs-base nil (emacs-lisp) elpa nil nil nil] [eieio-class-tag--cfgl-package evil-anzu spacemacs nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package evil-args spacemacs nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package evil-escape spacemacs-base nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package evil-evilified-state spacemacs-base nil nil local pre t nil] [eieio-class-tag--cfgl-package evil-exchange spacemacs nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package evil-iedit-state spacemacs nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package evil-indent-plus spacemacs nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package evil-jumper spacemacs nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package evil-lisp-state spacemacs nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package evil-matchit spacemacs nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package evil-mc spacemacs nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package evil-nerd-commenter spacemacs nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package evil-numbers spacemacs nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package evil-org org nil nil local nil nil nil] [eieio-class-tag--cfgl-package evil-search-highlight-persist spacemacs nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package evil-surround spacemacs-base nil (org) elpa nil nil nil] [eieio-class-tag--cfgl-package evil-terminal-cursor-changer spacemacs nil nil elpa nil nil t] [eieio-class-tag--cfgl-package evil-tutor spacemacs nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package evil-visualstar spacemacs-base nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package exec-path-from-shell spacemacs-base nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package expand-region spacemacs nil nil elpa nil nil nil] ...))
  configuration-layer//load-packages(([eieio-class-tag--cfgl-package ace-jump-helm-line spacemacs nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package ace-link spacemacs nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package ace-window spacemacs nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package adaptive-wrap spacemacs nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package aggressive-indent spacemacs nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package auto-compile emacs-lisp nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package auto-highlight-symbol spacemacs nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package avy spacemacs nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package bind-key spacemacs-base nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package bind-map spacemacs-base nil nil elpa pre nil nil] [eieio-class-tag--cfgl-package bookmark spacemacs-base nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package bracketed-paste spacemacs nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package buffer-move spacemacs nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package centered-cursor spacemacs nil nil local nil nil nil] [eieio-class-tag--cfgl-package clean-aindent-mode spacemacs nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package company nil nil (emacs-lisp) elpa nil nil nil] [eieio-class-tag--cfgl-package company-emoji nil nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package define-word spacemacs nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package desktop spacemacs nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package diminish spacemacs-base nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package doc-view spacemacs nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package ediff spacemacs-base nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package eldoc spacemacs-base nil (emacs-lisp) elpa nil nil nil] [eieio-class-tag--cfgl-package electric-indent-mode spacemacs-base nil nil built-in nil nil nil] [eieio-class-tag--cfgl-package elisp-slime-nav emacs-lisp nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package emacs-lisp emacs-lisp nil nil built-in nil nil nil] [eieio-class-tag--cfgl-package emoji-cheat-sheet-plus nil nil (org) elpa nil nil nil] [eieio-class-tag--cfgl-package eval-sexp-fu spacemacs nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package evil spacemacs-base nil (emacs-lisp) elpa nil nil nil] [eieio-class-tag--cfgl-package evil-anzu spacemacs nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package evil-args spacemacs nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package evil-escape spacemacs-base nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package evil-evilified-state spacemacs-base nil nil local pre t nil] [eieio-class-tag--cfgl-package evil-exchange spacemacs nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package evil-iedit-state spacemacs nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package evil-indent-plus spacemacs nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package evil-jumper spacemacs nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package evil-lisp-state spacemacs nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package evil-matchit spacemacs nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package evil-mc spacemacs nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package evil-nerd-commenter spacemacs nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package evil-numbers spacemacs nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package evil-org org nil nil local nil nil nil] [eieio-class-tag--cfgl-package evil-search-highlight-persist spacemacs nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package evil-surround spacemacs-base nil (org) elpa nil nil nil] [eieio-class-tag--cfgl-package evil-terminal-cursor-changer spacemacs nil nil elpa nil nil t] [eieio-class-tag--cfgl-package evil-tutor spacemacs nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package evil-visualstar spacemacs-base nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package exec-path-from-shell spacemacs-base nil nil elpa nil nil nil] [eieio-class-tag--cfgl-package expand-region spacemacs nil nil elpa nil nil nil] ...))
  configuration-layer/sync()
  (if (not (version<= spacemacs-emacs-min-version emacs-version)) (message (concat "Your version of Emacs (%s) is too old. " "Spacemacs requires Emacs version %d or above.") emacs-version spacemacs-emacs-min-version) (load-file (concat user-emacs-directory "core/core-load-paths.el")) (require (quote core-spacemacs)) (spacemacs/init) (spacemacs/maybe-install-dotfile) (configuration-layer/sync) (spacemacs/setup-startup-hook) (require (quote server)) (if (server-running-p) nil (server-start)))
  eval-buffer(#<buffer  *load*> nil "/Users/joe/.emacs.d/init.el" nil t)  ; Reading at buffer position 1076
  load-with-code-conversion("/Users/joe/.emacs.d/init.el" "/Users/joe/.emacs.d/init.el" t t)
  load("/Users/joe/.emacs.d/init" t t)
  #[0 "\205\266	\306=\203\307\310Q\202?	\311=\204\307\312Q\202?\313\307\314\315#\203*\316\202?\313\307\314\317#\203>\320\321\322!D\nB\323\202?\316\324\325\324\211#\210\324=\203e\326\327\330\307\331Q!\"\325\324\211#\210\324=\203d\210\203\247\332!\333\232\203\247\334!\211\335P\336!\203\201\211\202\214\336!\203\213\202\214\314\262\203\245\337\"\203\243\340\341#\210\342\343!\210\266\f?\205\264\314\325\344\324\211#)\262\207" [init-file-user system-type delayed-warnings-list user-init-file inhibit-default-init inhibit-startup-screen ms-dos "~" "/_emacs" windows-nt "/.emacs" directory-files nil "^\\.emacs\\(\\.elc?\\)?$" "~/.emacs" "^_emacs\\(\\.elc?\\)?$" initialization format-message "`_emacs' init file is deprecated, please use `.emacs'" "~/_emacs" t load expand-file-name "init" file-name-as-directory "/.emacs.d" file-name-extension "elc" file-name-sans-extension ".el" file-exists-p file-newer-than-file-p message "Warning: %s is newer than %s" sit-for 1 "default"] 7]()
  command-line()
  normal-top-level()
```

**Comment by d12frosted** [comment]
It's hard to tell what's wrong. First of all - how have you installed your emacs? If via `brew`, please share full command. Also make sure that it's up to date with latest git head.

**Comment by George9000** [comment]
I had a working copy of spacemacs, but an earlier version, on the same Emacs build. The trouble started when I upgraded to spacemacs 0.105.14. Initially I figured it was some elpa issue, but even after the --insecure flag and even commenting out elpa.gnu.org in the core config, nothing changed. And now, even rolling back a version doesn't solve the issue. Very puzzling.

Emacs build
```
brew install git
brew install autoconf
brew install autogen
brew install automake

brew install libtiff
brew install libjpeg
brew install --universal libpng
brew install giflib

sudo ln -s /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk/usr/include/libxml2/libxml/ /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk/usr/include/libxml

git clone git://git.savannah.gnu.org/emacs.git

./autogen.sh
./configure --with-ns CFLAGS="-g -O0"   # thats a zero after the capital O
make bootstrap
make install

# from emacs build directory 
cd nextstep/
cp -R ./Emacs.app /Applications/
```

**Comment by George9000** [comment]
Issue #4385 raised a similar problem. I took the suggested fix.

In the `.spacemacs` file, under the section dotspacemacs/init, set this variable to nil: `dotspacemacs-elpa-https nil`  Despite having launched Emacs with the --insecure flag, I continued seeing errors with packages until this variable was set to nil.

Either the recurrent issues with elpa https should be fixed or, perhaps, the default for this variable in a new .spacemacs file should be `nil.`

Anyway, this fixed a frustrating and persistent issue.
